### DISCUSSION TITLE
Enabling fpolicy performance counter

### DISCUSSION THREAD

**Discussion by seb1597** [NetApp/harvest#1646] [Q&A] [answered]
Hello,

Can you direct me to documentation for adding fpolicy performance counter in harvest?
I only see information regarding fpolicy name or statue in harvest.

May be you know if it is planned to be added in a future harvest release?

Thanks in advance
Regards,
Sébastien

**Comment by Hardikl** [comment, marked as answer]
Hi @seb1597 

This is the documentation link for how to add new Zapi/ZapiPerf object templates : https://netapp.github.io/harvest/22.11/configure-templates/#creatingediting-templates

To know about the supported counters for fpolicy object in ZapiPerf, you could try this: 
` ./bin/zapi -p POLLER_NAME show counters --object fpolicy`

**Comment by luddite516** [comment]
Hi:

I just started looking into this myself.  We have been running Harvest for years using ZAPI, and we have customized it to collect fpolicy and some other counters.
For this year's upgrade, we are planning to transition harvest to using REST API.  So far, most counters/collections are working OK but it appears that OnTAP 9.14 does not have a REST interface for collecting fpolicy counters.

Our workaround is to continue to use ZAPI if to collect those metrics we cannot get through REST.

**Reply by rahulguptajss** [reply]
@luddite516 The performance counters utilized by Harvest's default templates have been implemented for REST by ONTAP team. As documented [here](https://netapp.github.io/harvest/24.11.1/architecture/rest-strategy/#can-i-use-the-rest-and-zapi-collectors-at-the-same-time), you can use both REST and ZAPI collectors simultaneously. However, once ZAPI is deprecated, this will pose a problem. Could you please submit a RFE with ONTAP to implement the same functionality for REST?
